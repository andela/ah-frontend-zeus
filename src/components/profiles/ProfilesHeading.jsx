import React, { Component } from 'react';

class ProfilesHeading extends Component {
  render() {
    return (
      <section id="dashboard-page" className="flex-grow-1">
        <section id="profile" className="mt-5 mb-2">
          <div className="container">
            <div className="row">
              <div className="col-md-10 m-auto">
                <h2 className="display-2" style={{ textAlign: 'center' }}>
                  {this.props.heading}
                  <input
                    style={{ display: this.props.displayFilter ? 'block' : 'none' }}
                    onChange={this.props.handleChange}
                    name="display"
                    placeholder="Filter Users..."
                    id="filter-users"
                  />
                </h2>
              </div>
            </div>
          </div>
        </section>
      </section>
    );
  }
}

export default ProfilesHeading;
