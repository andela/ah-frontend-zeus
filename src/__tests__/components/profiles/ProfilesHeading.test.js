import React from 'react';
import Enzyme from 'enzyme';
import EnzymeAdapter from 'enzyme-adapter-react-16';
import { shallow } from 'enzyme';
import ProfilesHeading from '../../../components/profiles/ProfilesHeading';

Enzyme.configure({ adapter: new EnzymeAdapter() });

describe('<ProfilesHeading />', () => {
  it('matches snapshot', () => {
    const wrapper = shallow(<ProfilesHeading />);
    expect(wrapper).toMatchSnapshot();
  });
});
