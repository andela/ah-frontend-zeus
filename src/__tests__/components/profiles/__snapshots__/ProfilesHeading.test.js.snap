// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProfilesHeading /> matches snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ProfilesHeading />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <section
        className="mt-5 mb-2"
        id="profile"
      >
        <div
          className="container"
        >
          <div
            className="row"
          >
            <div
              className="col-md-10 m-auto"
            >
              <h2
                className="display-2"
                style={
                  Object {
                    "textAlign": "center",
                  }
                }
              >
                <input
                  id="filter-users"
                  name="display"
                  placeholder="Filter Users..."
                  style={
                    Object {
                      "display": "none",
                    }
                  }
                />
              </h2>
            </div>
          </div>
        </div>
      </section>,
      "className": "flex-grow-1",
      "id": "dashboard-page",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="container"
        >
          <div
            className="row"
          >
            <div
              className="col-md-10 m-auto"
            >
              <h2
                className="display-2"
                style={
                  Object {
                    "textAlign": "center",
                  }
                }
              >
                <input
                  id="filter-users"
                  name="display"
                  placeholder="Filter Users..."
                  style={
                    Object {
                      "display": "none",
                    }
                  }
                />
              </h2>
            </div>
          </div>
        </div>,
        "className": "mt-5 mb-2",
        "id": "profile",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="row"
          >
            <div
              className="col-md-10 m-auto"
            >
              <h2
                className="display-2"
                style={
                  Object {
                    "textAlign": "center",
                  }
                }
              >
                <input
                  id="filter-users"
                  name="display"
                  placeholder="Filter Users..."
                  style={
                    Object {
                      "display": "none",
                    }
                  }
                />
              </h2>
            </div>
          </div>,
          "className": "container",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="col-md-10 m-auto"
            >
              <h2
                className="display-2"
                style={
                  Object {
                    "textAlign": "center",
                  }
                }
              >
                <input
                  id="filter-users"
                  name="display"
                  placeholder="Filter Users..."
                  style={
                    Object {
                      "display": "none",
                    }
                  }
                />
              </h2>
            </div>,
            "className": "row",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <h2
                className="display-2"
                style={
                  Object {
                    "textAlign": "center",
                  }
                }
              >
                <input
                  id="filter-users"
                  name="display"
                  placeholder="Filter Users..."
                  style={
                    Object {
                      "display": "none",
                    }
                  }
                />
              </h2>,
              "className": "col-md-10 m-auto",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  undefined,
                  <input
                    id="filter-users"
                    name="display"
                    placeholder="Filter Users..."
                    style={
                      Object {
                        "display": "none",
                      }
                    }
                  />,
                ],
                "className": "display-2",
                "style": Object {
                  "textAlign": "center",
                },
              },
              "ref": null,
              "rendered": Array [
                undefined,
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "id": "filter-users",
                    "name": "display",
                    "onChange": undefined,
                    "placeholder": "Filter Users...",
                    "style": Object {
                      "display": "none",
                    },
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "input",
                },
              ],
              "type": "h2",
            },
            "type": "div",
          },
          "type": "div",
        },
        "type": "div",
      },
      "type": "section",
    },
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <section
          className="mt-5 mb-2"
          id="profile"
        >
          <div
            className="container"
          >
            <div
              className="row"
            >
              <div
                className="col-md-10 m-auto"
              >
                <h2
                  className="display-2"
                  style={
                    Object {
                      "textAlign": "center",
                    }
                  }
                >
                  <input
                    id="filter-users"
                    name="display"
                    placeholder="Filter Users..."
                    style={
                      Object {
                        "display": "none",
                      }
                    }
                  />
                </h2>
              </div>
            </div>
          </div>
        </section>,
        "className": "flex-grow-1",
        "id": "dashboard-page",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="container"
          >
            <div
              className="row"
            >
              <div
                className="col-md-10 m-auto"
              >
                <h2
                  className="display-2"
                  style={
                    Object {
                      "textAlign": "center",
                    }
                  }
                >
                  <input
                    id="filter-users"
                    name="display"
                    placeholder="Filter Users..."
                    style={
                      Object {
                        "display": "none",
                      }
                    }
                  />
                </h2>
              </div>
            </div>
          </div>,
          "className": "mt-5 mb-2",
          "id": "profile",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="row"
            >
              <div
                className="col-md-10 m-auto"
              >
                <h2
                  className="display-2"
                  style={
                    Object {
                      "textAlign": "center",
                    }
                  }
                >
                  <input
                    id="filter-users"
                    name="display"
                    placeholder="Filter Users..."
                    style={
                      Object {
                        "display": "none",
                      }
                    }
                  />
                </h2>
              </div>
            </div>,
            "className": "container",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <div
                className="col-md-10 m-auto"
              >
                <h2
                  className="display-2"
                  style={
                    Object {
                      "textAlign": "center",
                    }
                  }
                >
                  <input
                    id="filter-users"
                    name="display"
                    placeholder="Filter Users..."
                    style={
                      Object {
                        "display": "none",
                      }
                    }
                  />
                </h2>
              </div>,
              "className": "row",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <h2
                  className="display-2"
                  style={
                    Object {
                      "textAlign": "center",
                    }
                  }
                >
                  <input
                    id="filter-users"
                    name="display"
                    placeholder="Filter Users..."
                    style={
                      Object {
                        "display": "none",
                      }
                    }
                  />
                </h2>,
                "className": "col-md-10 m-auto",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    undefined,
                    <input
                      id="filter-users"
                      name="display"
                      placeholder="Filter Users..."
                      style={
                        Object {
                          "display": "none",
                        }
                      }
                    />,
                  ],
                  "className": "display-2",
                  "style": Object {
                    "textAlign": "center",
                  },
                },
                "ref": null,
                "rendered": Array [
                  undefined,
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "id": "filter-users",
                      "name": "display",
                      "onChange": undefined,
                      "placeholder": "Filter Users...",
                      "style": Object {
                        "display": "none",
                      },
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "input",
                  },
                ],
                "type": "h2",
              },
              "type": "div",
            },
            "type": "div",
          },
          "type": "div",
        },
        "type": "section",
      },
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
